### DISCUSSION TITLE
How to convert DynamoDBEvent images to DynamoDBv2.Document ?

### DISCUSSION THREAD

**Discussion by Dreamescaper** [aws/aws-lambda-dotnet#1654] [Q&A] [answered]
I have a lambda subscribed to dynamoDb events. I want to convert event's records to regular JSON, currently we use something like that:
```csharp
public async Task FunctionHandler(DynamoDBEvent input, ILambdaContext context)
{
        foreach (var record in input.Records)
        {
                var image = record.Dynamodb.NewImage;
                var document = Document.FromAttributeMap(image);
                var json = document.ToJson();
                /*   ....    */
        }
}
```

However, it doesn't work after the https://github.com/aws/aws-lambda-dotnet/pull/1648 is merged.
Could anyone suggest what is the easiest way to do same after the update?

**Comment by dscpinheiro** [comment]
It's related to that change, we'll take a look.

Curious: What are you using the regular JSON for? Are you saving it to a separate DynamoDB table?

**Reply by Dreamescaper** [reply]
In one case we're pushing the JSON to SNS, and we want it to be a "regular" JSON, not DynamoDb entry JSON.

In couple of other cases we simply want to deserialize 'NewImage' to a DTO, but it's hard to do with DynamoDb JSON.
(Previously we could use `_dynamoDbContext.FromDocument<T>(document)`, which is not available now as well).

**Reply by dscpinheiro** [reply]
Got it, thanks for the explanation. We missed this use case when doing the major version bump.

I talked to @normj and we should create an alternative `ToJson` method in the new event class. I'll convert this to an issue so we can better track it.

If you weren't impacted by the `IsSet` methods in the `AttributeValue` class (which is the reason we removed the SDK dependency), you should be able to keep using version `2.3.0` of `Amazon.Lambda.DynamoDBEvents` (there are no other functional differences between that and `3.0.0`).

**Reply by CriggerMarg** [reply]
Can you please post link to the issue here so I can track it too?

**Reply by CriggerMarg** [reply]
Also, we are simulating dynamo db event in our unit tests and with 3.0.0 we are no longer able to use DynamoDBContext.ToDocument(DTO).ToAttributeMap() method as another AttributeValue class is used in dynamoDbEvent.

**Reply by dscpinheiro** [reply, marked as answer]
@CriggerMarg The corresponding issue is https://github.com/aws/aws-lambda-dotnet/issues/1657

**Reply by ashovlin** [reply]
Replied over on #1657 , but commenting here too that we just released version 3.1.0 which adds `ToJson`/`ToJsonPretty` which can be used with `DynamoDBEvent`. This allows one to convert the event to JSON, which can then be used to convert again to the SDK's document type then an object persistence class.
